global_controller/global_controller.py

```diff
@@ -0,0 +1,181 @@
+from common.configurable_model import ConfigurableModel
+from bit_vector import BitVector
+from enum import Enum
+import magma as m
+import fault
+
+
+class GC_reg_addr(Enum):
+    TST_addr = 0
+    stall_addr = 1
+    clk_sel_addr = 2
+    rw_delay_sel_addr = 3
+    clk_switch_delay_sel_addr = 4
+
+
+class GC_op(Enum):
+    NOP = 0
+    config_write = 1
+    config_read = 2
+    write_A050 = 4
+    write_TST = 5
+    read_TST = 6
+    global_reset = 7
+    write_stall = 8
+    read_stall = 9
+    advance_clk = 10
+    read_clk_domain = 11
+    switch_clk = 12
+    write_rw_delay_sel = 13
+    read_rw_delay_sel = 14
+    write_clk_switch_delay_sel = 15
+    read_clk_switch_delay_sel = 16
+
+
+def gen_global_controller(config_data_width: int,
+                          config_addr_width: int,
+                          config_op_width: int):
+
+    class _GlobalController(ConfigurableModel(32, 32)):
+        def __init__(self):
+            super().__init__()
+            self.NUM_STALL_DOMAINS = 4
```

Comment:
is this an instance variable or global const? It is formatted as a global const but used as an instance variable. If it is the latter (which I think it is, it should be in lower case).